src/extension.ts

```diff
@@ -84,6 +109,114 @@ function previewSelection() {
 	previewCode(selectedText);
 }
 
+/**
+ * Runs the `manimgl` command in the terminal, with the current cursor's line number:
+ * manimgl <file_name> <ClassName> [-se <lineNumber>]
+ * 
+ * - Saves the active file.
+ * - Previews the scene at the cursor's line (end of line)
+ * - If the cursor is on a class definition line, then `-se <lineNumber>`
+ *   is NOT added, i.e. the whole scene is previewed.
+ * - (3b1b's version also copies this command to the clipboard with additional
+ *   args `--prerun --finder -w`. We don't do that here.)
+ */
+async function startScene() {
+	const editor = vscode.window.activeTextEditor;
+	if (!editor) {
+		vscode.window.showErrorMessage(
+			'No opened file found. Please place your cursor at a line of code.'
+		);
+		return;
+	}
+
+	// Save active file
+	vscode.commands.executeCommand('workbench.action.files.save');
+
+	const languageId = editor.document.languageId;
+	if (languageId !== 'python') {
+		vscode.window.showErrorMessage("You don't have a Python file open.");
+		return;
+	}
+
+	const lines = editor.document.getText().split("\n");
+
+	// Find which lines define classes
+	// E.g. here, classLines = [{ line: "class FirstScene(Scene):", index: 3 }, ...]
+	const classLines = lines
+		.map((line, index) => ({ line, index }))
+		.filter(({ line }) => /^class (.+?)\((.+?)\):/.test(line));
+
+	const cursorLine = editor.selection.start.line;
```

Comment:
If you make a selection, don't you rather want that the _end_ of your selection is treated as the current line of the cursor?

Reply:
could you please elaborate?
Maybe with an example?